htslib/cram.h

```diff
@@ -167,9 +218,15 @@ HTSLIB_EXPORT
 int32_t cram_block_get_crc32(cram_block *b);
 HTSLIB_EXPORT
 void *  cram_block_get_data(cram_block *b);
-
 HTSLIB_EXPORT
 enum cram_content_type cram_block_get_content_type(cram_block *b);
+HTSLIB_EXPORT
+enum cram_block_method cram_block_get_method(cram_block *b);
+
+HTSLIB_EXPORT
+enum cram_block_method cram_expand_method(uint8_t *data, int32_t size,
+                                          enum cram_block_method comp,
+                                          cram_method_details *cm);
```

Comment:
It would be useful to design this interface so `cram_method_details` could be expanded if necessary in the future (e.g. to better describe future codecs).  Currently it's difficult because the size of the `cram_method_details` struct is set by the caller, which locks in the ABI.  If instead we made this function allocate `cram_method_details` struct then it would be possible to safely add extra fields on the end without breaking old code.

Alternatively we could wrap up the real function so that we can pass on information about the interface the caller expects, e.g.:
```c
#define cram_method_details_vers 1
static inline enum cram_block_method cram_expand_method(uint8_t *data, int32_t size,
                                          enum cram_block_method comp,
                                          cram_method_details *cm) {
    HTSLIB_EXPORT
    extern enum cram_block_method cram_expand_method_(uint8_t *data, int32_t size,
                                              enum cram_block_method comp,
                                              cram_method_details *cm,
                                              int method_details_version);
    return cram_expand_method_(data, size, comp, cm, cram_method_details_vers);
}
```
or possibly like this:
```c
static inline enum cram_block_method cram_expand_method(uint8_t *data, int32_t size,
                                          enum cram_block_method comp,
                                          cram_method_details *cm) {
    HTSLIB_EXPORT
    extern enum cram_block_method cram_expand_method1(uint8_t *data, int32_t size,
                                              enum cram_block_method comp,
                                              cram_method_details *cm);
    return cram_expand_method1(data, size, comp, cm);
}
```
where we make `cram_expand_method2()` should the struct change, but leave `cram_expand_method1()` exported for the benefit of callers expecting the original version ... or we use symbol versioning (which this sort-of is), but I don't think that works on some platforms we support.

Reply:
The versioning here is something that we don't bother with anywhere else.  The traditional htslib way of doing it is simply to add a new function with 2, 3, etc after it.  Eg `hgetc2`, `bcf_index_build3`, `sam_index_load3`, `kgetline2` etc.  We also have structures with numerics such as bam_pileup1_t`, but as yet haven't needed to create a new structure name.

I prefer this as the documentation is then always in sync with the API.  If we silently change the API and use versioning shenanigans to keep the ABI consistent, then there's a risk people use old API documentation that is not in sync with the library.  Explicit versioning via differing function / structure name is much easier to reliably document.

I think just cross the bridge when we come to it by adding a new function / structure name, as it's not likely to happen any time soon given these fields are based on the CRAM specification contents.